### PR TITLE
add recipe for Graphite Powder from Graphite Blocks

### PR DESCRIPTION
**Pull request by mspielberg**
Per @Repofme1:
`*Also graphite block sounces & sinks (carbon-processing tech)`

### CODE REVIEW

File: migrations/1-3-2_1-3-3.lua [hunk 1]
```diff
@@ -0,0 +1,9 @@
+for index, force in pairs(game.forces) do
```
Comment (jodokus31) [inline migrations/1-3-2_1-3-3.lua]: `game.reload_script()` missing?
Surround by
```
if recipes["graphite-grinding"] and technologies["carbon-processing"]
..
end
```
?

File: prototypes/recipe/bulk/bulk.lua [hunk 2]
```diff
@@ -1450,6 +1450,17 @@ data:extend(
 	ingredients = {{"coke", 4}},
 	result = "graphite-block"
 },
+--Graphite Grinding
```
Comment (jodokus31) [inline prototypes/recipe/bulk/bulk.lua]: Good idea! Then you only need to ship graphite blocks, which are more dense and grind it to powder on-side

File: migrations/1-3-2_1-3-3.lua (hunk 1 above)
Comment (mspielberg) [reply migrations/1-3-2_1-3-3.lua]: In theory, we're not doing anything special with the rocket-silo that would require the scenario script to be reloaded. And migration scripts are run in order, so these items should exist. Note that none of the migration scripts in base check for nil. There was definitely a bug with the migrations scripts before, I'll try to have a look and see if I can determine the cause.

File: prototypes/recipe/bulk/bulk.lua (hunk 2 above)
Comment (mspielberg) [reply prototypes/recipe/bulk/bulk.lua]: More importantly, the basic recipes gives 8x coke -> 4x powder, and then 4x powder -> 1x block. The electric furnace recipe in Carbon Processing gives 4x coke -> 1x block, so adding this transformation saves 50% of your input coke.

File: migrations/1-3-2_1-3-3.lua (hunk 1 above)
Comment (jodokus31) [reply migrations/1-3-2_1-3-3.lua]: The problem was, that the migration wants to change a recipe, which gots deleted in the current version. When someone adds XM mod to the game, then it runs all migrations, but the access to the deleted recipe fails.
Regarding script reload, i've got so much too learn with modding :)

**Comment by mspielberg** [comment]
Superseded by mainline change.

**Pull request closed without merging**